lib/interface_quda.cpp

```diff
@@ -9178,116 +9103,55 @@ void calcMG_loop_wOneD_TSM_wExact(void **gaugeToPlaquette,
-	
-	//HP solve
-	//-------------------------------------------------
-	SolverParam solverParam(*param);
-	Solver *solve = Solver::create(solverParam, m, mSloppy, mPre, 
-				       profileInvert);
-	(*solve)   (*out,*in);
-	delete solve;
-	dirac.reconstruct(*x,*b,param->solution_type);
-	//-------------------------------------------------
-	
-	
-	//LP solve
-	//-------------------------------------------------
-	double orig_tol = param->tol;
-	long int orig_maxiter = param->maxiter;
-	// Set the low-precision criterion
-	if(TSM_maxiter==0) param->tol = TSM_tol;
-	else if(TSM_tol==0) param->maxiter = TSM_maxiter;  
-	
-	// Create the low-precision solver
-	SolverParam solverParam_LP(*param);
-	Solver *solve_LP = Solver::create(solverParam_LP, m, mSloppy, 
-					  mPre, profileInvert);
-	(*solve_LP)(*out,*in);
-	delete solve_LP;
-	dirac.reconstruct(*x_LP,*b,param->solution_type);
-	
-	// Revert to the original, high-precision values
-	if(TSM_maxiter==0) param->tol = orig_tol;           
-	else if(TSM_tol==0) param->maxiter = orig_maxiter;      
-	//-------------------------------------------------
-	
-	sol    = new cudaColorSpinorField(*x);
-	sol_LP = new cudaColorSpinorField(*x_LP);
-      }
-
+      //HP solve
+      //-------------------------------------------------
+      SolverParam solverParam(*param);
+      Solver *solve = Solver::create(solverParam, m, mSloppy, mPre, 
+				     profileInvert);
+      (*solve)   (*out,*in);
+      delete solve;
+      dirac.reconstruct(*x,*b,param->solution_type);
+      //-------------------------------------------------
+      
+      
+      //LP solve
+      //-------------------------------------------------
+      double orig_tol = param->tol;
+      long int orig_maxiter = param->maxiter;
+      // Set the low-precision criterion
+      if(TSM_maxiter==0) param->tol = TSM_tol;
+      else if(TSM_tol==0) param->maxiter = TSM_maxiter;  
+      
+      // Create the low-precision solver
+      SolverParam solverParam_LP(*param);
+      Solver *solve_LP = Solver::create(solverParam_LP, m, mSloppy, 
+					mPre, profileInvert);
+      (*solve_LP)(*out,*in);
+      delete solve_LP;
+      dirac.reconstruct(*x_LP,*b,param->solution_type);
+      
+      // Revert to the original, high-precision values
+      if(TSM_maxiter==0) param->tol = orig_tol;           
+      else if(TSM_tol==0) param->maxiter = orig_maxiter;      
+      //-------------------------------------------------
+      
+      sol    = new cudaColorSpinorField(*x);
+      sol_LP = new cudaColorSpinorField(*x_LP);
+      
       for(int dstep=0;dstep<nDeflSteps;dstep++){
 	int NeV_defl = loopInfo.deflStep[dstep];
 	printfQuda("# Performing TSM contractions for NeV = %d\n",NeV_defl);
-
+	
 	if(smethod==0){
 	  t1 = MPI_Wtime();
 	  K_vector->downloadFromCuda(sol,flag_eo);
 	  K_vector->download();
 	  deflation->projectVector(*K_vecdef,*K_vector,is+1,NeV_defl);
-	  // HP Solution is projected and put into x, x <- (1-UU^dag) x
+
```

Comment:
Again, the following is a misleading comment. It's not any exact part that is projected out of the solution. The solution itself (the spinor "x") gets projected with (1-P), as the original comment stated.